src/output-json-stats.c

```diff
@@ -265,20 +265,29 @@ json_t *StatsToJSON(const StatsTable *st, uint8_t flags)
         for (x = 0; x < st->ntstats; x++) {
             uint32_t offset = x * st->nstats;
 
+            // Stats for for this thread.
+            json_t *thread = json_object();
+            if (unlikely(thread == NULL)) {
+                json_decref(js_stats);
+                json_decref(threads);
+                return NULL;
+            }
+
             /* for each counter */
             for (u = offset; u < (offset + st->nstats); u++) {
                 if (st->tstats[u].name == NULL)
                     continue;
 
+                // Seems this holds, but assert in debug builds.
+                assert(strcmp(st->tstats[u].tm_name, st->tstats[u].tm_name) == 0);
```

Comment:
Ewh, this was meant to be
```suggestion
                assert(strcmp(st->tstats[offset].tm_name, st->tstats[u].tm_name) == 0);
```